proposals/0021/README.md

```diff
@@ -23,8 +23,137 @@ No-Goal:
 
 ## Content
 
+Tinkerbell uses workflows to describe the work that has to be done in a piece of
+hardware. A workflow is composed of a set of actions, the smallest unit of work
+inside a workflow.
+
+A workflow is a descriptive specification shaped as a YAML file; actions are OCI
+or Docker images run as container via Docker.
+
+A key concept for Tinkerbell is reusability. Action is written, tested, and
+built once, and they can be reused across many workflows.
+
+Reusability improves reliability and stability because more users will stress
+and push the limit of a specific workflow or Action. The community wants to
+build a healthy mechanism to share workflows and actions.
+
+As part of this proposal, we will stay focused on Action because, as I wrote, it
+is the smallest unit of work, and it is packaged as an OCI image, a very well
+known and established packaging format.
+
+Having a well-established way to share actions and workflows is essential
+because domain experts will share their experience. Operating system experts,
+for example, will be able to maintain and share their Actions and Workflows, all
+the community will benefit from that.
+
+Actions look like this: Wipe a disk Partitioning a disk Interact with a BMC to
+restart a server Sending a message on Slack at some point during workflow
+execution
+
+The Action Hub will work as a catalog for Actions first and later for Workflows.
+
+The Tinkerbell community wants to stay independent, and we identified the
+[ArtifactHub](https://artifacthub.io/) project sponsored by CNCF (as Tinkerbell
+itself) as the right partner for building the Action Hub. I shared our
+requirements and idea during an ArtifactHub contributors' meeting, and they are
+happy to collaborate with us.
+
+I identified a couple of needs: We need a place (a git repository) where actions
+can be built and shared in collaboration. This is the source of truth and the
+place where contributors collaborate on all the Actions the community wants to
+support. Via CI/CD, we can validate, build, and push images to a registry.  We
+need an image registry where our Actions will be released to. We have to find
+one; it can be Quay, Docker Hub, our one, or something else. I think it is
```

Comment:
Intrigued on your thoughts on which would be best. Hosting out own would allow us more functionality and control.. We'd also avoid any rate limiting.. I also wonder how long Quay will be free for.

Reply:
Yeah, I would like to run our own one, it looks better and "forever" but we need a bit of structure, I hope @rawkode will have time to setup the infra for us so we can start experiment with something easier.

I noticed that ArtifactHub accept more than one image, so we can specify multi repositories, that should help.
I started with quay.io/tinkerbell-actions. Just to keep things going :)